### PR TITLE
[BEAM-14068]sklearn inference example and it test

### PR DESCRIPTION
**Pull request by AnandInguva**
**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by AnandInguva** [comment]
@ryanthompson591 can you take a look on the example and see if this is good way to include prebuilt dataset into the pipeline instead of reading from GCS?

**Comment by AnandInguva** [comment]
Run Python 3.8 PostCommit

File: sdks/python/apache_beam/examples/inference/sklearn_inference_example.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  def process(self, element):
+    true_label, prediction_result = element
+    prediction = prediction_result.inference
+    yield true_label, prediction
+
+
+def setup_pipeline(options: PipelineOptions, args=None):
+  """Sets up Sklearn RunInference pipeline"""
+  model_loader = SklearnModelLoader(
+      model_file_type=ModelFileType.PICKLE, model_uri=args.model_path)
```
Comment (ryanthompson591) [inline sdks/python/apache_beam/examples/inference/sklearn_inference_example.py:55]: pickle is the default, so no need to specify.

File: sdks/python/apache_beam/ml/inference/sklearn_inference_it_test.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    extra_opts = {
+        'dataset': 'fetch_20newsgroups',
+        'output': output_file,
+        'model_path': model_path,
+    }
+
+    sklearn_inference_example.run(
+        test_pipeline.get_full_options_as_args(**extra_opts),
+        save_main_session=False)
+    self.assertEqual(FileSystems().exists(output_file), True)
```
Comment (ryanthompson591) [inline sdks/python/apache_beam/ml/inference/sklearn_inference_it_test.py:49]: Is it possible to read anything from the file and validate it?

File: sdks/python/apache_beam/ml/inference/sklearn_inference_it_test.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import uuid
+
+from apache_beam.io.filesystems import FileSystems
+from apache_beam.examples.inference import sklearn_inference_example
+from apache_beam.testing.test_pipeline import TestPipeline
+
+
+class SklearnInference(unittest.TestCase):
+  @pytest.mark.it_postcommit
+  @pytest.mark.sickbay_direct
```
Comment (ryanthompson591) [inline sdks/python/apache_beam/ml/inference/sklearn_inference_it_test.py:32]: Why do we want to sickbay spark/flink/direct.  Shouldn't those all work?

Or will they fail when data is hosted on gcs?

File: sdks/python/apache_beam/examples/inference/sklearn_inference_example.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+def parse_known_args(argv):
+  """Parses args for the workflow."""
+  parser = argparse.ArgumentParser()
+  parser.add_argument(
+      '--dataset',
+      dest='dataset',
+      help='Sklearn built-in dataset name',
+      # datasets in sklearn.datasets module
+      choices=['olivetti_faces', 'fetch_20newsgroups'])
```
Comment (ryanthompson591) [inline sdks/python/apache_beam/examples/inference/sklearn_inference_example.py:81]: I suppose this is fine, but IMO it would be better to use a hosted dataset.

File: sdks/python/apache_beam/examples/inference/sklearn_inference_example.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+          subset='test', data_home='/tmp')
+      for x, y in zip(newsgroups_test['data'], newsgroups_test['target']):
+        # (n_feat,) is supported but not (1, n_feat)
+        yield y, np.squeeze(x.toarray())
+    else:
+      raise NotImplementedError
+
+
+class PostProcessor(beam.DoFn):
+  """Post process PredictionResult to output true_label and
```
Comment (ryanthompson591) [inline sdks/python/apache_beam/examples/inference/sklearn_inference_example.py:44]: Maybe rename true_label to expected_label or just label. ML people should know what that is.

File: sdks/python/apache_beam/examples/inference/sklearn_inference_example.py [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+import apache_beam as beam
+
+from apache_beam.ml.inference.api import RunInference
+from apache_beam.ml.inference.sklearn_inference import ModelFileType
+from apache_beam.ml.inference.sklearn_inference import SklearnModelLoader
+from apache_beam.options.pipeline_options import PipelineOptions
+from apache_beam.options.pipeline_options import SetupOptions
+
+
+class GetData(beam.DoFn):
```
Comment (ryanthompson591) [inline sdks/python/apache_beam/examples/inference/sklearn_inference_example.py:30]: I think this would be a stronger example if it followed a more conventional approach like loading from a csv.

Also in this case, even having labels is not important. Since I suppose in a true example if the labels were already known it wouldn't be valuable.

Plus I don't think the labels are really a good key, later when we validate data. Because I don't think the data is guaranteed to come out ordered the same as it was input.

A better key for later validating expectation might be just the index of the example.

File: sdks/python/apache_beam/examples/inference/sklearn_inference_example.py (hunk 5 above)
Comment (AnandInguva) [reply sdks/python/apache_beam/examples/inference/sklearn_inference_example.py:29]: conventional way, we call label as true_label. Anyway, I am gonna change the docstring on this method soon

File: sdks/python/apache_beam/examples/inference/sklearn_inference_example.py (hunk 1 above)
Comment (AnandInguva) [reply sdks/python/apache_beam/examples/inference/sklearn_inference_example.py:46]: Yes, but I thought it would be nice just to pass it as a param :)
